ibverbs/src/lib.rs

```diff
@@ -199,7 +192,7 @@ impl<'devlist> Device<'devlist> {
     ///  - `ENOMEM`: Out of memory (from `ibv_query_port_attr`).
     ///  - `EMFILE`: Too many files are opened by this process (from `ibv_query_gid`).
     ///  - Other: the device is not in `ACTIVE` or `ARMED` state.
-    pub fn open(&self) -> io::Result<Context> {
+    pub fn open(&self) -> io::Result<Arc<Context>> {
```

Comment:
This change is suspicious to me. I don't think we should mandate that callers use an `Arc`. It's true that it makes the lifetimes go away since we can ensure that the `Context` lives for as long as we want, but does having the lifetime constraints actually post a problem for writing real code? Just having the `Context` live for "however long it needs to" _feels_ wrong somehow.

Reply:
That's a fair point. I'll look into making the lifetimes more explicit instead

Reply:
I think they were all correct the way they were, so I'm mostly curious why the decision was made to remove them. Was there a particular use that became too cumbersome?

Reply:
My assumption is that the original author removed this out of ease of use. Looking through it I should have reviewed this whole file more carefully, since the one big important thing added were the device attributes